lib/acme/client/resources/challenges/base.rb

```diff
@@ -8,6 +8,10 @@ def initialize(client, **arguments)
     assign_attributes(arguments)
   end
 
+  def [](key)
+    to_h[key]
+  end
+
```

Comment:
I'm not sure why not simply calling `#to_h` within your app for that use case?

Also, would you prefer to keep them as a challenge object? I could implement an `UnknownChallenge`  and this way you could call `status`, `request_validation`, etc, directly on it?